app/templates/play/level/modal/progress-view.jade

```diff
@@ -40,43 +44,82 @@
+              p
+                b(data-i18n="play_level.share_this_project")
+          if view.session.isFake()
+            .row
+              .col-xs-8.col-xs-offset-2
+                .alert.alert-warning(data-i18n="sharing.unavailable")
           else
-            span(data-i18n='sharing.share_web')
-        p
-          span(data-i18n='sharing.victory_course_share_prefix')
-          span= ' '
-          a(href=view.shareURL, target='_blank')
+            .row
+              .col-sm-6.col-sm-offset-2
+                input.text-h4.semibold.form-control#share-level-input(value=view.shareURL)
+              .col-sm-2
+                button#share-level-btn.btn.btn-illustrated.btn-primary
+                  span(data-i18n='sharing.copy_url')
+        .well.well-sm.well-parchment
+          .row.publish-text
+            .col-xs-8.col-xs-offset-2
+              if view.session.get('published')
+                p
+                  b(data-i18n="play_level.already_published_prefix")
+                  br
+                  span(data-i18n="play_level.already_published_suffix")
+              else
+                p
+                  b(data-i18n="play_level.ready_to_share")
+                  br
+                  span(data-i18n="play_level.click_publish")
+      else
+        .well.well-sm.well-parchment
+          h3.text-uppercase
             if view.level.isType('game-dev')
-              span(data-i18n='sharing.victory_course_share_game')
+              span(data-i18n='sharing.share_game')
             else
-              span(data-i18n='sharing.victory_course_share_web')
-          span= ' '
-          span(data-i18n='sharing.victory_course_share_suffix')
-        if view.session.isFake()
-          .alert.alert-warning(data-i18n="sharing.unavailable")
-        else
-          .row
-            .col-sm-9
-              input.text-h4.semibold.form-control.input-lg#share-level-input(value=view.shareURL)
-            .col-sm-3
-              button#share-level-btn.btn.btn-lg.btn-success.btn-illustrated
-                span(data-i18n='sharing.copy_url')
+              span(data-i18n='sharing.share_web')
+          p
+            span(data-i18n='sharing.victory_course_share_prefix')
+            span= ' '
+            a(href=view.shareURL, target='_blank')
+              if view.level.isType('game-dev')
+                span(data-i18n='sharing.victory_course_share_game')
+              else
+                span(data-i18n='sharing.victory_course_share_web')
+            span= ' '
+            span(data-i18n='sharing.victory_course_share_suffix')
+          if view.session.isFake()
+            .alert.alert-warning(data-i18n="sharing.unavailable")
+          else
+            .row
+              .col-sm-10
+                input.text-h4.semibold.form-control#share-level-input(value=view.shareURL)
+              .col-sm-2
+                button#share-level-btn.btn.btn-illustrated
+                  span(data-i18n='sharing.copy_url')
```

Comment:
Gosh this modal is overloaded. Maybe going forward, having separate components would be better?